calico_node/startup/startup_test.go

```diff
@@ -202,44 +216,35 @@ var _ = Describe("FV tests against a real etcd", func() {
 				log.Println("Error creating client:", err)
 			}
 
-			// Set combination of env vars expected to cause terminate.
-			os.Setenv("NO_DEFAULT_POOLS", "true")
-			os.Setenv("CALICO_IPV4POOL_CIDR", "172.16.0.0/16")
+			// Set the env variables specified.
+			for _, env := range envList {
+				os.Setenv(env.key, env.value)
+			}
 
 			// Run the UUT.
 			configureIPPools(c)
 
-			It("should have terminated", func() {
-				Expect(exitCode).To(Equal(1))
-			})
-		})
-	})
-
-	Describe("Bad IP Pool value", func() {
-		exitCode = 0
-		oldExit := exitFunction
-		exitFunction = fakeExitFunction
-		defer func() { exitFunction = oldExit }()
-
-		// Erase etcd clean.
-		testutils.CleanEtcd()
-
-		// Create a new client.
-		c, err := testutils.NewClient("")
-		if err != nil {
-			log.Println("Error creating client:", err)
-		}
-
-		// Set bad IP Pool string.
-		os.Setenv("CALICO_IPV4POOL_CIDR", "172.16.0.0a/16")
-
-		// Run the UUT.
-		configureIPPools(c)
+			Expect(my_ec).To(Equal(1))
+		},
 
-		It("should have terminated", func() {
-			Expect(exitCode).To(Equal(1))
-		})
-	})
+		Entry("Bad IPv4 Pool CIDR", []EnvItem{{"CALICO_IPV4POOL_CIDR", "172.16.0.0a/24"}}),
+		Entry("Too small IPv4 Pool CIDR", []EnvItem{{"CALICO_IPV4POOL_CIDR", "172.16.0.0/27"}}),
```

Comment:
Ah yes you are quite right the CIDR check was already in place.  I think now, though, since we have moved the validation check into the Create() method in the API - we need to update glide to pick up the libcalico-go changes I think.  Is that not the case?